ui/src/routes/devices.$id.tsx

```diff
@@ -62,27 +66,14 @@ interface CloudLoaderResp {
   } | null;
 }
 
-export type AuthMode = "password" | "noPassword" | null;
 export interface LocalDevice {
   authMode: AuthMode;
   deviceId: string;
 }
 
 const deviceLoader = async () => {
-  const res = await api
-    .GET(`${DEVICE_API}/device/status`)
-    .then(res => res.json() as Promise<DeviceStatus>);
-
-  if (!res.isSetup) return redirect("/welcome");
-
-  const deviceRes = await api.GET(`${DEVICE_API}/device`);
-  if (deviceRes.status === 401) return redirect("/login-local");
-  if (deviceRes.ok) {
-    const device = (await deviceRes.json()) as LocalDevice;
-    return { authMode: device.authMode };
-  }
-
-  throw new Error("Error fetching device");
+  const device = await checkAuth();
```

Comment:
This won't work as the `checkAuth()` -> `checkDeviceAuth()` can return a redirect. As far as I can understand, you can just replace the returns with throws, and it should work.